src/dapla_metadata/datasets/model_backwards_compatibility.py

```diff
@@ -187,6 +191,98 @@ def _cast_to_date_type(value_to_update: str | None) -> str | None:
     )
 
 
+def convert_is_personal_data(supplied_metadata: dict[str, Any]) -> None:
+    """Convert 'is_personal_data' values in the supplied metadata to boolean.
+
+    Iterates over variables in the supplied metadata and updates the
+    'is_personal_data' field:
+      - Sets it to True for NON_PSEUDONYMISED_ENCRYPTED_PERSONAL_DATA and PSEUDONYMISED_ENCRYPTED_PERSONAL_DATA.
+      - Sets it to False for NOT_PERSONAL_DATA.
+
+    Args:
+        supplied_metadata: The metadata dictionary to be updated.
+    """
+    for variable in supplied_metadata["datadoc"]["variables"]:
+        value = variable["is_personal_data"]
+        if value in (
+            "NON_PSEUDONYMISED_ENCRYPTED_PERSONAL_DATA",
+            "PSEUDONYMISED_ENCRYPTED_PERSONAL_DATA",
+        ):
+            variable["is_personal_data"] = True
+        elif value == "NOT_PERSONAL_DATA":
+            variable["is_personal_data"] = False
+
+
+def copy_pseudonymization_metadata(supplied_metadata: dict[str, Any]) -> None:
+    """Copies pseudonymization metadata from the old pseudonymization section into the corresponding variable.
```

Comment:
There's one more field we should migrate here, `supplied_metadata["pseudonymization"]["pseudo_dataset"]["dataset_pseudo_time"]`. The value can be saved in `pseudonymization_time` for each variable.